Source/Particles/Collision/BinaryCollision/PhotonPhoton/LinearBreitWheelerCrossSection.H

```diff
@@ -0,0 +1,61 @@
+/* Copyright 2021 Neil Zaim
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+
+#ifndef LINEAR_BREIT_WHEELER_CROSS_SECTION_H
+#define LINEAR_BREIT_WHEELER_CROSS_SECTION_H
+
+#include "Utils/WarpXConst.H"
+
+#include <AMReX_REAL.H>
+
+#include <cmath>
+
+/**
+ * \brief Computes the total proton-boron fusion cross section. When E_kin_star < 3.5 MeV, we use
+ * the analytical fits given in W.M. Nevins and R. Swain, Nuclear Fusion, 40, 865 (2000). When
+ * E_kin_star > 3.5 MeV, we use a simple power law fit of the data presented in Buck et al.,
+ * Nuclear Physics A, 398(2), 189-202 (1983). Both fits return the same value for
+ * E_kin_star = 3.5 MeV.
+ *
+ * @param[in] E_kin_star the kinetic energy of the proton-boron pair in its center of mass frame,
+ * in SI units.
+ * @return The total cross section in SI units (square meters).
+ */
+AMREX_GPU_HOST_DEVICE AMREX_INLINE
+amrex::ParticleReal LinearBreitWheelerCrossSection (const amrex::ParticleReal& E_star)
+{
+    using namespace amrex::literals;
+
+    constexpr amrex::ParticleReal c_sq = PhysConst::c * PhysConst::c;
+    constexpr amrex::ParticleReal me = PhysConst::m_e;
+    // If kinetic energy is 0, return a 0 cross section and avoid later division by 0.
+    if (E_star < 2._prt*me*c_sq) {return 0._prt;}
+
+    constexpr auto one_half_pr = amrex::ParticleReal(1./2.);
+    constexpr amrex::ParticleReal pi = MathConst::pi;
+    constexpr amrex::ParticleReal re2 = PhysConst::r_e * PhysConst::r_e;
+    auto constexpr pow2 = [](double const x) { return x*x; };
+    auto constexpr pow4 = [](double const x) { return x*x*x*x; };
```

Comment:
Is there a reason for restricting these to `double const x` instead of `auto const x` ?

Reply:
I don't have a strong preference, but using `auto` possibly hides an information that could be useful for another developer. What do you think?

Reply:
I would  suggest using `double` only if the method should always use double precision (and I would suggest adding a comment to explain why this is done, if this is the case). Otherwise, using `auto` would allow the calculation to be performed in single precision is WarpX is compiled in single precision.